demos/LowAllocationWebServer/HttpHeaders.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Text.Utf8;
+using LowAllocationServer;
+
+namespace System.Net.Http.Buffered
+{
+    public struct HttpHeaders : IEnumerable<KeyValuePair<Utf8String, Utf8String>>
+    {
+        private readonly ByteSpan _bytes;
+        
+        public HttpHeaders(ByteSpan bytes)
+        {
+            _bytes = bytes;            
+        }                
+
+        public Utf8String this[string headerName]
+        {
+            get
+            {
+                foreach (var header in this)
+                {
+                    if (header.Key == headerName)
+                    {
+                        return header.Value;
+                    }
+                }
+
+                return Utf8String.Empty;
```

Comment:
That's a very good point. Do you have a preference here? We could return a Utf8String? or throw.